### PR TITLE
fix(www): broken link on home page

### LINKED ISSUE #12306: 404 link on Gatsby home page
## Description

Clicking the "Creating Plugins" button on the https://www.gatsbyjs.org/ home page in the plugins section (see image) leads to a 404.

image

Refreshing the page after that triggers the correct(?) redirect.

Additionally, there is at least one more broken link on this page:
https://www.gatsbyjs.org/docs/plugins/
and there could be other cases so maybe global search for the slug to search and replace is a good idea.

### Steps to reproduce

1. Go to live site home page (incognito or not)
2. Scroll down to the plugins section
3. Click the "Creating Plugins" button

### Expected result

We should see the relevant page or be redirected correctly.

### Actual result

We see a 404.

### Environment

N/A

### PR DESCRIPTION
**Pull request by jumpalottahigh**
## Description

There is a broken link on the gatsbyjs.org home page, as described in #12306 .
This PR replaces all occurrences of the broken link.

## Related Issues

Closes #12306

### CODE REVIEW

**Comment by marcysutton** [comment]
I think that page was redirected to this one, but it isn't working in that context for some reason: https://www.gatsbyjs.org/docs/how-plugins-work/

Perhaps that URL would be a better fit than going to the source plugin tutorial?

**Comment by jumpalottahigh** [comment]
Yep, you are right @marcysutton , there is a redirect setup in `gatsby-node.js` but as you mentioned it doesn't work.

Sure I can update the PR to your suggestion if you think that's better. My logic was that the button reads "Creating plugins" and I would expect something more actionable, and more directly connected to making plugins rather than a page with more information about plugins. Also the "Using plugins" button next to the broken button already provides a link with more information about plugins.

Let me know what you think and if I should update the PR.

**Comment by marcysutton** [comment]
The first link in the set goes to `/plugins`, which lists all the available plugins. The How Plugins Work page I recommended for the second link starts with the text "You may be looking to build a plugin that doesn’t exist yet," and comes in the docs information architecture before `/docs/source-plugin-tutorial/`, so it seems like a good spot for it in my opinion...especially since not all plugins are source plugins necessarily. The third link goes to the `/docs/plugins` landing page, and that one is focused more on using plugins versus building them. What do you think?

Thanks again for working on this PR!

File: docs/blog/2018-03-07-why-we-created-the-plugin-library/index.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ author: Shannon Soper
 tags: ["plugins", "documentation"]
 ---
 
-Earlier this week, we launched a [Plugin Library](/packages/) on Gatsbyjs.org and are excited for you to use it! This article explains how UX research drove the design of this library and the accompanying [Plugin Authoring](/docs/plugin-authoring/) doc.
+Earlier this week, we launched a [Plugin Library](/packages/) on Gatsbyjs.org and are excited for you to use it! This article explains how UX research drove the design of this library and the accompanying [Plugin Authoring](/docs/source-plugin-tutorial/) doc.
```
Comment (jumpalottahigh) [inline docs/blog/2018-03-07-why-we-created-the-plugin-library/index.md]:
```suggestion
Earlier this week, we launched a [Plugin Library](/packages/) on Gatsbyjs.org and are excited for you to use it! This article explains how UX research drove the design of this library and the accompanying [Plugin Authoring](/docs/how-plugins-work/) doc.
```

File: docs/blog/2018-03-07-why-we-created-the-plugin-library/index.md [hunk 2]
```diff
@@ -79,7 +79,7 @@ Here are some ways you can help make the Gatsby plugin ecoystem great:
 
 - Share feedback on the plugin library on [GitHub Issue #4394](https://github.com/gatsbyjs/gatsby/issues/4394).
 - If you created a plugin and it's not showing up in the library, double check that the package has "gatsby-plugin" in its keywords.
-- Create plugins (or publish ones you've already built)! If you're interested, the [Plugin Authoring](/docs/plugin-authoring/) page can help.
+- Create plugins (or publish ones you've already built)! If you're interested, the [Plugin Authoring](/docs/source-plugin-tutorial/) page can help.
```
Comment (jumpalottahigh) [inline docs/blog/2018-03-07-why-we-created-the-plugin-library/index.md]:
```suggestion
- Create plugins (or publish ones you've already built)! If you're interested, the [Plugin Authoring](/docs/how-plugins-work/) page can help.
```

File: docs/blog/2018-05-11-six-reasons-i-chose-gatsby/index.md [hunk 3]
```diff
@@ -16,7 +16,7 @@ I've been working with React for the better part of 3.5 years. I know it. I'm ef
...
+One of the first things about Gatsby that impressed me was its plugin system. Like many other OSS tools such as Webpack and Babel, much of the power of Gatsby is provided by plugins. And that's a good thing. Its plugin architecture allows for incredibly deep integrations into almost every aspect of Gatsby: build configuration, data extraction and transformation, the build process, the browser at runtime, etc. This system has allowed not only the core Gatsby team to create powerful plugins, but the community to create equally powerful plugins as well. I've even tried my hand at [writing one](https://github.com/raygesualdo/gatsby-plugin-settings) (quite successfully, I might add). If there's a task you're looking to accomplish with Gatsby, odds are there is already a plugin for it. And if not, there's [plenty of documentation](/docs/source-plugin-tutorial/) to help you get started writing one.
```
Comment (jumpalottahigh) [inline docs/blog/2018-05-11-six-reasons-i-chose-gatsby/index.md]:
```suggestion
One of the first things about Gatsby that impressed me was its plugin system. Like many other OSS tools such as Webpack and Babel, much of the power of Gatsby is provided by plugins. And that's a good thing. Its plugin architecture allows for incredibly deep integrations into almost every aspect of Gatsby: build configuration, data extraction and transformation, the build process, the browser at runtime, etc. This system has allowed not only the core Gatsby team to create powerful plugins, but the community to create equally powerful plugins as well. I've even tried my hand at [writing one](https://github.com/raygesualdo/gatsby-plugin-settings) (quite successfully, I might add). If there's a task you're looking to accomplish with Gatsby, odds are there is already a plugin for it. And if not, there's [plenty of documentation](/docs/how-plugins-work/) to help you get started writing one.
```

File: docs/contributing/submit-to-plugin-library.md [hunk 4]
```diff
@@ -7,7 +7,7 @@ title: Submit to Plugin Library
 In order to add your plugin to the [Plugin Library](/plugins/), you need to:
 
 1. publish a package to npm (learn how [here](https://docs.npmjs.com/getting-started/publishing-npm-packages)),
-2. include the [required files](/docs/plugin-authoring/#what-files-does-gatsby-look-for-in-a-plugin) in your plugin code,
+2. include the [required files](/docs/source-plugin-tutorial/#what-files-does-gatsby-look-for-in-a-plugin) in your plugin code,
```
Comment (jumpalottahigh) [inline docs/contributing/submit-to-plugin-library.md]:
```suggestion
2. include the [required files](/docs/how-plugins-work/#what-files-does-gatsby-look-for-in-a-plugin) in your plugin code,
```

File: docs/docs/plugins.md [hunk 5]
```diff
@@ -19,7 +19,7 @@ Gatsby has a large and growing ecosystem of official and community plugins. To b
 
 ## Learn more about plugins
 
-For documentation with further detail on what comprises a Gatsby plugin (file structure, etc), see the [plugin authoring page](/docs/plugin-authoring/).
+For documentation with further detail on what comprises a Gatsby plugin (file structure, etc), see the [plugin authoring page](/docs/source-plugin-tutorial/).
```
Comment (jumpalottahigh) [inline docs/docs/plugins.md]:
```suggestion
For documentation with further detail on what comprises a Gatsby plugin (file structure, etc), see the [plugin authoring page](/docs/how-plugins-work/).
```

File: docs/docs/source-plugin-tutorial.md [hunk 6]
```diff
@@ -381,7 +381,7 @@ Your plugin has been adapted from Jason Lengstorf's [gatsby-source-pixabay plugi
 
 Try adding new features to your plugin, for example - download images from Pixabay, improve error handling, add documentation or automated tests.
 
-Check out Gatsby's docs on [plugin authoring](/docs/plugin-authoring/) and [creating a source plugin](/docs/create-source-plugin).
+Check out Gatsby's docs on [plugin authoring](/docs/source-plugin-tutorial/) and [creating a source plugin](/docs/create-source-plugin).
```
Comment (jumpalottahigh) [inline docs/docs/source-plugin-tutorial.md]:
```suggestion
Check out Gatsby's docs on [plugin authoring](/docs/how-plugins-work/) and [creating a source plugin](/docs/source-plugin-tutorial/).
```

File: docs/docs/styling.md [hunk 7]
```diff
@@ -3,7 +3,7 @@ title: Styling
 overview: true
 ---
 
-There are so many ways to add styles to your website -- and Gatsby supports almost every possible option, through official and community plugins. (_If there isn’t a plugin yet for your favorite option, consider [contributing one](/docs/plugin-authoring/)!_)
+There are so many ways to add styles to your website -- and Gatsby supports almost every possible option, through official and community plugins. (_If there isn’t a plugin yet for your favorite option, consider [contributing one](/docs/source-plugin-tutorial/)!_)
```
Comment (jumpalottahigh) [inline docs/docs/styling.md]:
```suggestion
There are so many ways to add styles to your website -- and Gatsby supports almost every possible option, through official and community plugins. (_If there isn’t a plugin yet for your favorite option, consider [contributing one](/docs/how-plugins-work/)!_)
```

File: docs/tutorial/part-four/index.md [hunk 8]
```diff
@@ -63,7 +63,7 @@ See the [Using Gatsby without GraphQL](/docs/using-gatsby-without-graphql/) guid
 If you're building a small site, one efficient way to build it is to pull in unstructured data as outlined in this guide, using `createPages` API, and then if the site becomes more complex later on, you move on to building more complex sites, or you'd like to transform your data, follow these steps:
 
 1.  Check out the [Plugin Library](/plugins/) to see if the source plugins and/or transformer plugins you'd like to use already exist
-2.  If they don't exist, read the [Plugin Authoring](/docs/plugin-authoring/) guide and consider building your own!
+2.  If they don't exist, read the [Plugin Authoring](/docs/source-plugin-tutorial/) guide and consider building your own!
```
Comment (jumpalottahigh) [inline docs/tutorial/part-four/index.md]:
```suggestion
2.  If they don't exist, read the [Plugin Authoring](/docs/how-plugins-work/) guide and consider building your own!
```

File: examples/using-local-plugins/README.md [hunk 9]
```diff
@@ -2,7 +2,7 @@
 
 This example demonstrates usage of a local plugin -- in this case a source plugin.
 
-You might also be interested in the docs section on [local plugins](/docs/plugin-authoring/#local-plugins), or the [source plugin tutorial](/docs/source-plugin-tutorial/).
+You might also be interested in the docs section on [local plugins](/docs/source-plugin-tutorial/#local-plugins), or the [source plugin tutorial](/docs/source-plugin-tutorial/).
```
Comment (jumpalottahigh) [inline examples/using-local-plugins/README.md]:
```suggestion
You might also be interested in the docs section on [local plugins](/docs/how-plugins-work/#local-plugins), or the [source plugin tutorial](/docs/source-plugin-tutorial/).
```

File: www/src/components/ecosystem/ecosystem-board.js [hunk 10]
```diff
@@ -51,7 +51,7 @@ class EcosystemBoard extends Component {
             { label: `Browse Plugins`, to: `/plugins/` },
             {
               label: `Creating Plugins`,
-              to: `/docs/plugin-authoring/`,
+              to: `/docs/source-plugin-tutorial/`,
```
Comment (jumpalottahigh) [inline www/src/components/ecosystem/ecosystem-board.js]:
```suggestion
              to: `/docs/how-plugins-work/`,
```

File: www/src/components/homepage/homepage-ecosystem.js [hunk 11]
```diff
@@ -175,7 +175,7 @@ const HomepageEcosystem = ({ featuredItems }) => (
           { label: `Browse Plugins`, to: `/plugins/` },
           {
             label: `Creating Plugins`,
-            to: `/docs/plugin-authoring/`,
+            to: `/docs/source-plugin-tutorial/`,
```
Comment (jumpalottahigh) [inline www/src/components/homepage/homepage-ecosystem.js]:
```suggestion
            to: `/docs/how-plugins-work/`,
```

File: www/src/pages/plugins.js [hunk 12]
```diff
@@ -107,7 +107,7 @@ class Plugins extends Component {
             blazing fast site even more awesome. If you
             {`'`}d like to create your own plugin, see the
             {` `}
-            <Link to="/docs/plugin-authoring/">Plugin Authoring</Link> page in
+            <Link to="/docs/source-plugin-tutorial/">Plugin Authoring</Link> page in
```
Comment (jumpalottahigh) [inline www/src/pages/plugins.js]:
```suggestion
            <Link to="/docs/how-plugins-work/">Plugin Authoring</Link> page in
```

**Comment by jumpalottahigh** [comment]
@marcysutton I think it makes very much sense. Thanks for explaining. I updated the links now as you suggested.

**Pull request merged**